espnet2/asr/transducer/beam_search_transducer.py

```diff
@@ -85,6 +88,9 @@ def __init__(
         self.decoder = decoder
         self.joint_network = joint_network
 
+        self.sos = sos
```

Comment:
Yes we can directly set it as `self.sos = vocab_size - 1`. Since the regular beam search was using this format, I did the same here. But keeping #4032 in mind (where the ASR model doesn't have the sos token as a parameter), it makes sense to set it directly.

Reply:
Yes that makes perfect sense!